pyaedt/generic/grpc_plugin_dll_class.py

```diff
@@ -0,0 +1,402 @@
+
+    def __getattr__(self, funcName):
+        try:
+            if funcName == "ScopeID":  # backward compatible for IronPython wrapper.
+                return self.objectID
+            return self.__GetObjMethod__(funcName)
+        except Exception:
+            raise GrpcApiError("Failed to get grpc API AEDT attribute {}".format(funcName))
+
+    def __setattr__(self, attrName, val):
+        if attrName == "objectID" or attrName == "__methodNames__":
+            raise GrpcApiError("Modify this attribute is not allowed")
+        elif attrName in self.__methodNames__:
+            raise GrpcApiError(attrName + " is a function name")
+        else:
+            super().__setattr__(attrName, val)
+
+    def __del__(self):
+        if "ReleaseAedtObject" in dir(self.dllapi):
+            self.dllapi.ReleaseAedtObject(self.objectID)
+
+    def match(self, patternStr):  # IronPython wrapper implemented this function return IEnumerable<string>.
+        class IEnumerable(list):
+            def __getattr__(self, key):
+                if key == "Count":
+                    return len(self)
+
+        pattern = re.compile(patternStr)
+        found = IEnumerable()
+        allMethods = self.__methodNames__
+        for method in allMethods:
+            if pattern.match(method):
+                found.append(method)
+        return found
+
+    def GetHashCode(self):  # IronPython build in function
+        return self.__hash__()
+
+
+class AedtPropServer(AedtObjWrapper):
+    def __init__(self, objID, listFuncs, aedtapi):
+        AedtObjWrapper.__init__(self, objID, listFuncs, aedtapi)
+        self.__dict__["__propMap__"] = None
+        self.__dict__["__propNames__"] = None
+
+    def __GetPropAttributes(self):
+
+        if self.__propMap__ == None:
+            propMap = {}
+            propNames = self.GetPropNames()
+            for prop in propNames:
+                attrName = ""
+                if prop[0].isdigit():
+                    attrName += "_"
+                for c in prop:
+                    if c.isalnum() == True:
+                        attrName += c
+                    else:
+                        attrName += "_"
+                propMap[attrName] = prop
+            self.__propMap__ = propMap
+        return self.__propMap__
+
+    def __dir__(self):
+        ret = super().__dir__().copy()
+        for attrName in self.__GetPropAttributes().keys():
+            ret.append(attrName)
+        return ret
+
+    def __getattr__(self, attrName):
+        try:
+            return super().__getattr__(attrName)
+        except AttributeError:
+            # if AedtAPI.IsAedtObjPropName(self.objectID, attrName, False):
+            #    return self.GetPropValue(attrName)
+            propMap = self.__GetPropAttributes()
+            if attrName in propMap:
+                return self.GetPropValue(propMap[attrName])
+            raise GrpcApiError("Failed to retrieve attribute {} from GRPC API".format(attrName))
```

Comment:
```suggestion
            raise GrpcApiError("Failed to retrieve attribute {} from gRPC API".format(attrName))
```
I don't know where to add the period.